x-pack/test/cloud_security_posture_functional/page_objects/add_cis_integration_form_page.ts

```diff
@@ -160,6 +175,11 @@ export function AddCisIntegrationFormPageProvider({
     await PageObjects.header.waitUntilLoadingHasFinished();
   };
 
+  const navigateToEditIntegrationPage = async () => {
+    const integrationList = await testSubjects.findAll('integrationNameLink');
+    await integrationList[0].click();
+  };
```

Comment:
the code will fail if element with `[data-test-subj='integrationNameLink']` is not DOM. `integrationList` can be empty and `Array Out of Bounds` wouldn't be helpful.

It is better to use `await testSubjects.click('integrationNameLink')`, this way WebDriver looks for the very first element in DOM matching locator (basically equal to `integrationList[0]`) but in case of failure with report that can't find element with specific locator.

Reply:
Okay nice to know. Thanks for the input